src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,48 @@
+package core.basesyntax;
+
+import core.basesyntax.figures.Circle;
+import core.basesyntax.figures.IsoscelesTrapezoid;
+import core.basesyntax.figures.Rectangle;
+import core.basesyntax.figures.RightTriangle;
+import core.basesyntax.figures.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+
+    private static final int QUANITY_OF_TYPES = 5;
+    private String color = ColorSupplier.getRandomColor();
```

Comment:
It will always be one color. You need to create it immediately before creating the figure

Reply:
not fixed, move to the `getRandomFigure()` method
create the instance of `ColorSupplier colorSupplier = new ColorSupplier();`